### PR TITLE
Remove some functions completions

### PR DESCRIPTION
**Pull request by j-bennet**
## Description

Fix for the lag problem in v2 (https://github.com/dbcli/pgcli/issues/979).

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [X] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by j-bennet** [comment]
Cleaned up and ready to merge. This does not fix the problem completely, but it's an improvement. There's also https://travis-ci.org/dbcli/pgcli/jobs/472740029 to address the same problem.

File: pgcli/packages/parseutils/meta.py [hunk 1]
```diff
@@ -79,6 +80,8 @@ def __init__(
         self.is_aggregate = is_aggregate
         self.is_window = is_window
         self.is_set_returning = is_set_returning
+        self.is_extension = bool(is_extension)
```
Comment (amjith) [inline pgcli/packages/parseutils/meta.py]: I'm having a hard time seeing how this is used.

Comment (j-bennet) [reply pgcli/packages/parseutils/meta.py]: Fixed in https://github.com/dbcli/pgcli/pull/982/commits/ed5dc2e68ac373d69a949f13be48ab20b4d2161f.

File: pgcli/pgcompleter.py [hunk 2]
```diff
@@ -636,22 +637,33 @@ def list_dict(pairs): # Turns [(a, b), (a, c)] into {a: [b, c]}
...
     def get_function_matches(self, suggestion, word_before_cursor, alias=False):
+
         if suggestion.usage == 'from':
             # Only suggest functions allowed in FROM clause
-            def filt(f): return not f.is_aggregate and not f.is_window
+
+            def filt(f):
+                return (not f.is_aggregate and
+                        not f.is_window and
+                        not f.is_extension and
```
[the remark this answers is not in the document]
Comment (j-bennet) [reply pgcli/pgcompleter.py]: I'll get you that info in a bit. There might have been a reason I left `is_extension` out of this PR.

**Comment by j-bennet** [comment]
Here are the results that I got when removing either extension functions, or non-public functions, or both, from completions:

https://gist.github.com/j-bennet/f6181e28460916910433882abb8d2b75

In my case (I have postgis installed)

* initially, I get up to 3603 completions
* removing extensions gets rid of 1059 completions
* removing non-public functions gets rid of 2374 completions
* removing both gets rid of 3434 completions.

I did not add a filter for underscored function names. It was not necessary, they went away once I filtered out extensions.

I handle a case when the user explicitly types a non-public schema name here:

https://github.com/dbcli/pgcli/blob/ed5dc2e68ac373d69a949f13be48ab20b4d2161f/pgcli/pgcompleter.py#L648

The one case I'm not handling here is filtering extensions. What if a user explicitly wants an extension function and starts typing it? What is a good way to handle that?

**Comment by j-bennet** [comment]
Going to merge this and deal with edge cases after.

**Pull request merged**